src/python/grpcio_tests/tests/interop/xds_interop_client.py

```diff
@@ -0,0 +1,201 @@
+            for peer, count in self._rpcs_by_peer.items():
+                response.rpcs_by_peer[peer] = count
+            response.num_failures = self._no_remote_peer + self._rpcs_needed
+        return response
+
+
+_global_lock = threading.Lock()
+_stop_event = threading.Event()
+_global_rpc_id: int = 0
+_watchers: Set[_StatsWatcher] = set()
+_global_server = None
+
+
+def _handle_sigint(sig, frame):
+    _stop_event.set()
+    _global_server.stop(None)
+
+
+class _LoadBalancerStatsServicer(test_pb2_grpc.LoadBalancerStatsServiceServicer
+                                ):
+
+    def __init__(self):
+        super(_LoadBalancerStatsServicer).__init__()
+
+    def GetClientStats(self, request: messages_pb2.LoadBalancerStatsRequest,
+                       context: grpc.ServicerContext
+                      ) -> messages_pb2.LoadBalancerStatsResponse:
+        print("Received stats request.")
+        sys.stdout.flush()
+        start = None
+        end = None
+        watcher = None
+        with _global_lock:
+            start = _global_rpc_id + 1
+            end = start + request.num_rpcs
+            watcher = _StatsWatcher(start, end)
+            _watchers.add(watcher)
+        response = watcher.await_rpc_stats_response(request.timeout_sec)
+        with _global_lock:
+            _watchers.remove(watcher)
+        return response
+
+
+# TODO: Accept finer-grained arguments.
+def _run_single_channel(args: argparse.Namespace):
+    global _global_rpc_id  # pylint: disable=global-statement
+    duration_per_query = 1.0 / float(args.qps)
+    with grpc.insecure_channel(args.server) as channel:
+        stub = test_pb2_grpc.TestServiceStub(channel)
+        while not _stop_event.is_set():
+            request_id = None
+            with _global_lock:
+                request_id = _global_rpc_id
+                _global_rpc_id += 1
+            print(f"Sending request to backend: {request_id}")
+            sys.stdout.flush()
+            start = time.time()
+            end = start + duration_per_query
+            call, _ = stub.UnaryCall.with_call(messages_pb2.SimpleRequest(),
+                                               timeout=float(
+                                                   args.rpc_timeout_sec))
+            print(f"Got result {request_id}")
+            sys.stdout.flush()
+            with _global_lock:
+                for watcher in _watchers:
+                    # TODO: Implement a peer details getter.
+                    peer = f"192.168.1.{request_id % 255}"
+                    watcher.on_rpc_complete(request_id, peer)
+            if args.print_response:
+                if call.code() == grpc.StatusCode.OK:
+                    print("Successful response.")
+                    sys.stdout.flush()
+                else:
+                    print(f"RPC failed: {call}")
+                    sys.stdout.flush()
+            now = time.time()
+            while now < end:
+                time.sleep(end - now)
```

Comment:
nit: Token bucket?

Reply:
Having a bursty traffic pattern would differ from the other clients and require a more complicated implementation. Unless there's a compelling reason for it, I'd prefer not to.

Reply:
Feel free to implement either way. I don't have a strong preference. A plain implementation can be:

```python
token = 1.0
last = time.monotonic()

while ...:
    if token < 1:
        time.sleep(1/args.qps * (1-token))
    now = time.monotonic()
    token = min(2, token - 1 + args.qps * (now-last))
    last = now
    ...
```

Reply:
Discussed with Eric. We need a steady QPS. That is, as close as possible to one RPC every `1 / QPS` seconds. That rules out a token bucket approach. Still, this is pretty cool. I wasn't aware of this algorithm. I'll have to keep this one in my back pocket.